internal/pkg/api/handleEnroll.go

```diff
@@ -252,6 +260,105 @@ func (et *EnrollerT) _enroll(
 				Msg("Error when trying to delete old agent with enrollment id")
 			return nil, err
 		}
+		// deleted, so clear the ID so code below knows it needs to be created
+		agent.Id = ""
+	}
+
+	var agentID string
+	if req.Id != nil && *req.Id != "" {
+		agentID = *req.Id
+
+		// Possible this agent already exists.
+		vSpan, vCtx := apm.StartSpan(ctx, "checkAgentID", "validate")
+		var err error
+		agent, err = dl.FindAgent(vCtx, et.bulker, dl.QueryAgentByID, dl.FieldID, agentID)
+		if err != nil {
+			zlog.Debug().Err(err).
+				Str("ID", agentID).
+				Msg("Agent with ID not found")
+			if !errors.Is(err, dl.ErrNotFound) && !strings.Contains(err.Error(), "no such index") {
+				vSpan.End()
+				return nil, err
+			}
+		} else if !agent.Active {
+			// inactive agent has been unenrolled and the API key has already been invalidated
+			// delete the current record as the new enrollment will overwrite this one
+			zlog.Debug().
+				Str("ID", agentID).
+				Msg("Inactive agent with ID found")
+			err = deleteAgent(ctx, zlog, et.bulker, agent.Id)
+			if err != nil {
+				zlog.Error().Err(err).
+					Str("AgentId", agent.Id).
+					Msg("Error when trying to delete old agent with same id")
+				return nil, err
```

Comment:
missing `vSpan.End()` right?

Reply:
we could extract to separate function that finds agent id with defer vspan.End to decrease probability of this in a future